daemon/buildinfo.c

```diff
@@ -200,8 +200,49 @@
 
 #define FEAT_YES_NO(x) ((x) ? "YES" : "NO")
 
+void buildinfo_read_install_type(void) {
```

Comment:
if possible without much trouble I would prefer this not to be duplicated with `analytics_get_install_type` as that is a maintainability issue

Reply:
I see two possible approaches we could take to achieve that:

1. Rename this function and move it to libnetdata, then change `analytics_get_install_type` to use this function and just copy the required data out of the produced environment variables.
2. Rewrite this to take a callback for the loop body, and have both the buildinfo code and the analytics code supply their own callbacks.

Not sure which would be the preferred approach here, but I do agree that this should be addressed.

Reply:
I prefer having one function in a place where both `buildinfo` and `analytics` can use it.
Not sure what exactly you mean with callback for loop body but that sounds like way too great overkill for such a simple functionality. If your trouble comes from difficulty returning 3 strings from 1 function ping me and I can help you.

Reply:
Resolving this as it’s largely done.